### ISSUE TITLE
editing tags is broken — throws exception in console

### ISSUE THREAD

**Issue by ericscheid** [naturalcrit/homebrewery#2532]
### Renderer

Both

### Browser

Chrome

### Operating System

MacOS

### What happened?

Attempting to commit a change to a tag in the metadata panel has no effect, and throws an error in console.

<img width="551" alt="image" src="https://user-images.githubusercontent.com/2605467/204833513-dfbdccc7-b57d-410a-9da1-3a9bb3f1f30c.png">

```
bundle.js:76373 Uncaught TypeError: e.persist is not a function
    at ReactClassComponent.handleFieldChange (bundle.js:76373:7)
    at Object.onChange (bundle.js:76611:23)
    at ReactClassComponent.handleChange (bundle.js:76945:16)
    at ReactClassComponent.addValue (bundle.js:76952:10)
    at onClick (bundle.js:77095:16)
    at Object.ba (bundle.js:30337:84)
    at ja (bundle.js:30337:238)
    at ka (bundle.js:30337:292)
    at oa (bundle.js:30338:72)
    at lc (bundle.js:30375:170)
```

Source offender is here:
https://github.com/naturalcrit/homebrewery/blob/3e44d0ad130d936a5b90c1ab5d4e1a6dcb770a1d/client/homebrew/editor/metadataEditor/metadataEditor.jsx#L55-L57

`event.persist()` is a react thing, so I have no suggestions on what to do here. This might even not be the true source cause .. it might be some other change that caused the e.persist() to no longer be valid in this context. All I know it has something to do with synthetic (react) events in async functions. Good luck.

**Comment by Gazook89** [comment]
So we are using React 16.14.0  (correct me if i'm wrong).  From Reacts docs:

> As of v17, e.persist() doesn’t do anything because the SyntheticEvent is no longer [pooled](https://reactjs.org/docs/legacy-event-pooling.html).

Do we ever expect to move to React v17?  If so, it's probably not worth the time to make `e.persist()` work for tags, and we should just work around it entirely.  In fact, I think moving to v17 would just remove the necessity for it entirely and fix this issue.  My understanding (which doesn't mean much) is that React <16 doesn't really hold onto SyntheticEvents into nested functions.   Using `persist()` enables using the event in these deeper functions.   When doing the Field Validation PR, I felt that I needed it for this to work (from metadataEditor.jsx):

```js
if(validationErr.length === 0){
	e.target.setCustomValidity('');   // without e.persist() this does not work, because 'e' is no longer linked to the original event
	this.props.onChange({
		...this.props.metadata,
		[name] : e.target.value
	});
```

The issue with the tags is happening because `e` isn't an event, so `e.persist()` isn't a function.  I'm not sure why `e` isn't an event, but I'm assuming it's because the `<input>` is inside an external component?  

This might all be a bunch of nonsense, but regardless I'm not likely to figure it out quickly though I will try.  But if there was any chance we'd move to React v17, then maybe it's not worth the bother.  Alternatively, adapting my Combobox for the tags entry might be an easy fix, but that might be even messier.

**Comment by calculuschild** [comment]
We can update React at any time as long as we make any accompanying changes so we don't break stuff. I'm assuming most react versions are more or less backwards compatible.

**Comment by Gazook89** [comment]
Will do.  For my notes, here is the React post about React 17.  https://reactjs.org/blog/2020/10/20/react-v17.html

It starts off with a hopeful tone:
> The React 17 release is unusual because it doesn’t add any new developer-facing features. Instead, this release is primarily focused on making it easier to upgrade React itself.
> ...
> We’ve postponed other changes until after React 17. The goal of this release is to enable gradual upgrades. If upgrading to React 17 were too difficult, it would defeat its purpose.

**Comment by Gazook89** [comment]
I updated to React 17 and don't seem to have any issues....I haven't tried to implement any changes to fix *this* issue, though it wasn't solved by just the upgrade.  But i can navigate around the site fine and haven't had any issues yet.

**Comment by calculuschild** [comment]
Ok. Do you want to go ahead and make a PR to update react and we can move forward assuming we will have at least react 17?

**Comment by Gazook89** [comment]
Honestly it's probably a good idea to have someone else give it a shot, to be sure I didn't miss something when it was installing?  There were 'warn' messages as I installed it, which I mostly took to be "not important" and powered through.  So...

**Comment by calculuschild** [comment]
Ok I'll do that one and see how it goes.

**Comment by calculuschild** [comment]
@Gazook89 I updated react to v17. I had to do a little finagling in the background on a couple other things but it seems to be working now...

**Comment by calculuschild** [comment]
Fixed by #2547
